In [1]:
# import required libraries

%pylab inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
from statsmodels.graphics.api import abline_plot
import patsy
import seaborn as sns
sns.set(context='notebook', style='whitegrid', palette='deep', font='sans-serif', font_scale=1, rc=None)
import urllib2 as url

Populating the interactive namespace from numpy and matplotlib


In [2]:
#Importing tha data required
data = pd.read_csv('bk_logit.csv')

In [4]:
#Printing data details and statistics
print data.info()
print data.describe()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 30799 entries, 0 to 30798
Data columns (total 4 columns):
bk             30799 non-null int64
age            30799 non-null int64
unit_size      30799 non-null float64
single_unit    30799 non-null int64
dtypes: float64(1), int64(3)
memory usage: 1.2 MB
None
                 bk           age     unit_size   single_unit
count  30799.000000  30799.000000  30799.000000  30799.000000
mean       0.307802     65.888211    218.766696      0.424267
std        0.461592     31.248772    124.524345      0.494239
min        0.000000      0.000000      9.290300      0.000000
25%        0.000000     50.000000    138.239670      0.000000
50%        0.000000     75.000000    187.292450      0.000000
75%        1.000000     87.000000    263.380000      1.000000
max        1.000000    212.000000    996.384700      1.000000


In [5]:
# performing a logisicc regression on bk due to age, size and single uniit
logit_mod = smf.logit('bk ~ age + unit_size + single_unit', data = data).fit()
print
print(logit_mod.summary())

Optimization terminated successfully.
         Current function value: 0.555319
         Iterations 6

                           Logit Regression Results                           
Dep. Variable:                     bk   No. Observations:                30799
Model:                          Logit   Df Residuals:                    30795
Method:                           MLE   Df Model:                            3
Date:                Sat, 10 Oct 2015   Pseudo R-squ.:                  0.1005
Time:                        09:16:36   Log-Likelihood:                -17103.
converged:                       True   LL-Null:                       -19013.
                                        LLR p-value:                     0.000
                  coef    std err          z      P>|z|      [95.0% Conf. Int.]
-------------------------------------------------------------------------------
Intercept      -2.2461      0.049    -46.114      0.000        -2.342    -2.151
age             0.0186   

In [6]:
#Calculating marginal effects table
marginal = logit_mod.get_margeff()
print(marginal.summary())

        Logit Marginal Effects       
Dep. Variable:                     bk
Method:                          dydx
At:                           overall
                 dy/dx    std err          z      P>|z|      [95.0% Conf. Int.]
-------------------------------------------------------------------------------
age             0.0035    8.2e-05     42.534      0.000         0.003     0.004
unit_size       0.0004      2e-05     18.723      0.000         0.000     0.000
single_unit    -0.1558      0.005    -29.286      0.000        -0.166    -0.145


Due to the results of the marginal effects of the logistic regression we can interpret that:

- The age has a positive effect on the probability of a sales in brooklyn by an increase of 0.35% for each year the property has.
- The size unit has also a positive effect on the probability of a sales in brooklyn of 0.04% for each additional square meter the unit has.
- The sales indicator of a single unit however has a negative effect on the probability of a sales of a unit in brooklyn by 15.58%.
    

We can confirm this interpretation by running  a OLS regression on BK using the same explanatory variables to see if the marginal effect is similar or not

In [8]:
# Calculating a OLS reggrerion on BK due to age, size, and single unit indicator
mod = smf.ols(formula='bk ~ age + unit_size + single_unit', data = data).fit()
print(mod.summary())

                            OLS Regression Results                            
Dep. Variable:                     bk   R-squared:                       0.117
Model:                            OLS   Adj. R-squared:                  0.117
Method:                 Least Squares   F-statistic:                     1364.
Date:                Sat, 10 Oct 2015   Prob (F-statistic):               0.00
Time:                        09:29:53   Log-Likelihood:                -17970.
No. Observations:               30799   AIC:                         3.595e+04
Df Residuals:                   30795   BIC:                         3.598e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [95.0% Conf. Int.]
-------------------------------------------------------------------------------
Intercept       0.0537      0.009      6.293      

As we can see from the coefficients of the OLS regression, the marginal effects estimated are almos identical in the logistic regression